.gitlab/common/shared.yml

```diff
@@ -35,6 +35,24 @@
   - *get_artifactory_token_win
   - if (($Env:USE_CACHING_PROXY_PYTHON -eq "true") -and ($Env:ARTIFACTORY_BYPASS -eq "false")) { $PIP_INDEX_URL="https://${Env:ARTIFACTORY_USERNAME}:${ARTIFACTORY_TOKEN}@${Env:ARTIFACTORY_URL}/${Env:ARTIFACTORY_PYPI_PATH}" }
 
+# Sets up a cache for gems used by Omnibus
+# Usage:
+# !reference [.cache_omnibus_ruby_deps, setup] somewhere ahead of invoking bundlex
+# !reference [.cache_omnibus_ruby_deps, cache] under `cache` for the same job
+.cache_omnibus_ruby_deps:
+  setup:
+    - pushd omnibus && bundle config set --local path 'vendor/bundle' && popd
+  cache:
+    key:
+      files:
+        - omnibus/Gemfile
+        # This is coarser than needed, but there's no more convenient way
+        # to get a hold of OMNIBUS_RUBY_VERSION and OMNIBUS_SOFTWARE version
+        - release.json
+      prefix: omnibus-deps-$CI_JOB_NAME
```

Comment:
Is there native code involved that would require us to version the cache per-job instead of having a general one?
I was expecting something more similar to our go deps cache, which is ran once at the beginning of the pipeline, without the need for each job to update their respective caches

Reply:
I just realized you addressed this in the PR description :facepalm: Sorry

Reply:
There _may_ be ruby dependencies with native code. So even if we're applying this only to Linux, there's a chance that some compatibility could arise. Definitely something that could be revised as a follow up (maybe tags can be used for the cache key? I don't know if that's possible).